component/header.go

```diff
@@ -105,42 +102,37 @@ func (u *updatesHeading) Render() *vecty.HTML {
 	)
 }
 
+func (u *updatesHeading) updateAll(*vecty.Event) {
+	// TODO.
+	fmt.Println("UpdateAll()")
+	js.Global.Get("UpdateAll").Invoke() // TODO: Do this via action?
+}
+
 func (u *updatesHeading) updateAllButton() *vecty.HTML {
-	if !u.UpdateSupported {
```

Comment:
I just deduplicated this element code into the case below.  If this element was rendered frequently, and the matched case below changed often, I'd also consider always rendering this as an anchor tag, and changing the other properties instead, so that the whole element doesn't need to be replaced.  As this is not the case, I left the case and element change as is, but this is an example of a potentially expensive operation.

Reply:
I think it's interesting how some of these changes that have performance as a goal are at odds with readability and ideal organization. I'm pretty sure the original code is a lot easier to read, the new version feels more like hand-written assembly for this very specific situation. Changing this component to have different rendering behavior would be harder.

Reply:
Because users are not railroaded into writing output as blocks of text, as they are in React, users should follow certain patterns if they want to avoid code that results in DOM element churn.  Perhaps I didn't go far enough here in reorganising the code, or explanation, to make it clear what the optimal patterns are.

This code shouldn't return two completely different elements, for example.